src/main/java/de/tum/in/www1/artemis/service/TutorialGroupService.java

```diff
@@ -85,6 +98,188 @@ public Set<StudentDTO> registerMultipleStudents(TutorialGroup tutorialGroup, Set
         return notFoundStudentDTOs;
     }
 
+    /**
+     * Import registrations
+     * <p>
+     * Important to note: A registration must contain a title of the tutorial group, but it must not contain a student.
+     *
+     * <ul>
+     *     <li> Only title -> Create Tutorial Group with the given title if it not exists
+     *     <li> Title and student -> Create Tutorial Group with given title if not exists AND register student.
+     *     <ul>
+     *         <li> If student is already registered in a tutorial group of the same course, the student will be deregistered from the old tutorial group.
+     *     </ul>
+     * </ul>
+     *
+     * @param course        The course to import the registrations for.
+     * @param registrations The registrations to import.
+     * @return The set of registrations with information about the success of the import
+     */
+    public Set<TutorialGroupRegistrationImportDTO> importRegistrations(Course course, Set<TutorialGroupRegistrationImportDTO> registrations) {
+        // container that will be filled with the registrations that could not be imported during the import process
+        var failedRegistrations = new HashSet<TutorialGroupRegistrationImportDTO>();
+
+        // === Step 1: Try to find all tutorial groups with the mentioned title. Create them if they do not exist yet ===
+        var registrationsWithTitle = filterOutWithoutTitle(registrations, failedRegistrations);
+        var tutorialGroupTitleToTutorialGroup = findOrCreateTutorialGroups(course, registrationsWithTitle).stream().collect(Collectors.groupingBy(TutorialGroup::getTitle));
+
+        // === Step 2: If the registration contains a student, try to find a user in the database with the mentioned registration number ===
+        var registrationWithUserIdentifier = registrationsWithTitle.stream().filter(registration -> {
+            if (registration.student() == null) {
+                return false;
+            }
+            boolean hasRegistrationNumber = registration.student().getRegistrationNumber() != null && !registration.student().getRegistrationNumber().isBlank();
+            boolean hasLogin = registration.student().getLogin() != null && !registration.student().getLogin().isBlank();
```

Comment:
StringUtils has a method that checks exactly the conditions in these lines and we use it in some other places

Reply:
I did not know about this! Nice that we have this! Thanks for the tip! Exchanged it for the Utils method in e06bf7bec608b82112281beffa48c9fd97da2d00